profiling/src/internal/observation/observations.rs

```diff
@@ -157,53 +166,18 @@ mod test {
         let t1 = Some(Timestamp::new(1).unwrap());
         let t2 = Some(Timestamp::new(2).unwrap());
 
-        o.add(s1, None, vec![1, 2, 3]);
-        o.add(s1, None, vec![4, 5, 6]);
-        o.add(s2, None, vec![7, 8, 9]);
-        o.iter().for_each(|(k, ts, v)| {
-            assert!(ts.is_none());
-            if k == s1 {
-                assert_eq!(v, vec![5, 7, 9]);
-            } else if k == s2 {
-                assert_eq!(v, vec![7, 8, 9]);
-            } else {
-                panic!("Unexpected key");
-            }
-        });
-        // Iter twice to make sure there are no issues doing that
-        o.iter().for_each(|(k, ts, v)| {
-            assert!(ts.is_none());
-            if k == s1 {
-                assert_eq!(v, vec![5, 7, 9]);
-            } else if k == s2 {
-                assert_eq!(v, vec![7, 8, 9]);
-            } else {
-                panic!("Unexpected key");
-            }
-        });
-        o.add(s3, t1, vec![10, 11, 12]);
+        o.add(s1, None, vec![1, 2, 3]).unwrap();
+        o.add(s1, None, vec![4, 5, 6]).unwrap();
+        o.add(s2, None, vec![7, 8, 9]).unwrap();
+        o.add(s3, t1, vec![10, 11, 12]).unwrap();
+        o.add(s2, t2, vec![13, 14, 15]).unwrap();
 
```

Comment:
Would be good to also check `o.len()` to make sure we actually added the samples

Reply:
Fixed in cd5b156302c751dd1cc7b90d99beb522abeaa936